hash/megadriv.xml

```diff
@@ -26682,8 +26708,25 @@ Black screen, writes in ROM area then jumps to PC=1a2000 (encryption?)
 		</part>
 	</software>
 
+	<software name="sonic2p12" cloneof="sonic2">
+		<!--
+			Earlier than "Beta 4" build.
+			Scene release with cracktro, no official unaltered build of this prototype has been found.
+			Dump originally in smd format, converted to raw binary due to softlists not supporting smd files.
+		-->
+		<description>Sonic The Hedgehog 2 (prototype, pirate)</description>
+		<year>1992</year>
+		<publisher>Sega</publisher>
+		<info name="release" value="19921109"/>
+		<part name="cart" interface="megadriv_cart">
+			<dataarea name="rom" width="16" endianness="big" size="1048576">
+				<rom name="sonic2.bin" size="1048576" crc="0de5d90b" sha1="cc29b1e9325f16cb0e66660129e861e743dab071"/>
+			</dataarea>
+		</part>
+	</software>
```

Comment:
I don’t think we want to add defaced ROMs like this.

Reply:
Are pirate games excluded now?

Other options are:
- Add a different defaced version that eliminates some of the pirate's defacing, making it a fake good version of it.
- MAME never has this build in it.

Reply:
You forgot the third option:
* Wait until a proper dump turns up.

These things eventually surface.  Even `raflesiau` showed up again.

Reply:
It's obviously supposed to be a placeholder until a proper dump turns up.

Reply:
But we generally don’t add defaced dumps because we don’t want to encourage them.

Reply:
Well, then just slip these into that "generally" because not even Sega's backup CDs had a build of this thing.

Reply:
You can load it as loose media, but we’re not adding a cracktro.

Reply:
Vas, there's lots of cracktros in other softlists and this one has a reason to be added.

Reply:
The other cracktros should be removed.  We don’t want to turn into the ironically named “GoodSets” that are full of garbage.  If we start adding defaced dumps, it will encourage people to only release defaced dumps to get their intros into the MAME set for self-promotion.

Reply:
...Regarding the last part of the previous reply, yeah, being something that people played on their real hardware could make for another reason to consider its inclusion. But having no known alternatives to it should already make it worth adding.

Perhaps there should be a separate "Licensed by Sega" softlist and have this one for everything else?

In any case, I think that any hypothetical cases of "self-promotion loopholes" should be discussed outside this very specific PR.

Reply:
If I'm gathering right `.smd` is a file format that comes with Super Magic Drive, if we want to support that then the files should be put in a separate SW list and as a specific Sonic & Knuckles cart slot.

Reply:
Can MAME read .smd files from softlists? I remember the game failing to boot when adding it before converting it, even if it loaded fine from the File Manager.

Reply:
I don't think there's *any* SMD support in MAME. Roms in that format are interleaved and they'd have to be decoded into a more sane format before being usable.

Reply:
wtf

Reply:
Oh wait, it does work if you load it in a slot instead of the software list? Huh, my bad.

Reply:
If it being a scene release is why this PR isn't being merged, drx and Evilhamwizard released a "cleaned" version on Hidden Palace in 2019 with the sha1 `9bebfb78049e7556fbef30f4a43d204fbf4c64f0`. If @cuavas prefers, maybe this version should be included with the scene release documented in the comments.

This approach is how No-Intro did documentation (before games were redumped) in the GBA times.

Reply:
I've just replaced the scene release with the provided cleaned up version.